pkgs/misc/emulators/pcem/default.nix

```
@@ -0,0 +1,30 @@
+{ stdenv, fetchzip, wxGTK31, coreutils, SDL2, openal, alsaLib, pkg-config
+, autoreconfHook, withNetworking ? true, withALSA ? true, ... }:
+
+let inherit (stdenv.lib) optional;
+
+in stdenv.mkDerivation rec {
+  pname = "pcem";
+  version = "17";
+
+  buildInputs = [ wxGTK31 coreutils SDL2 openal ] ++ optional withALSA alsaLib;
+  nativeBuildInputs = [ autoreconfHook pkg-config ];
+
+  src = fetchzip {
+    url = "https://pcem-emulator.co.uk/files/PCemV${version}Linux.tar.gz";
+    stripRoot = false;
+    sha256 = "067pbnc15h6a4pnnym82klr1w8qwfm6p0pkx93gx06wvwqsxvbdv";
+  };
+
+  configureFlags = [ "--enable-release-build" ]
+    ++ optional withNetworking "--enable-networking"
```

```suggestion
  configureFlags = [ "--enable-release-build" "--enable-networking" ]
```

Not sure why we want this flag.

---

This allows you to disable emulation of any network cards and the in-tree libslirp build. I know I wouldn't want to put any of these VMs on my network accidentally, and debated if this should be default true or default false.

---

This should be a runtime setting but fine with me.